src/generic_solver.cpp

```diff
@@ -868,7 +1007,29 @@ Result GenericSolver::check_sat()
 
 Result GenericSolver::check_sat_assuming(const TermVec & assumptions)
 {
-  assert(false);
+  string names;
+  for (Term t : assumptions)
+  {
+    // assumptions can only be Boolean literals
+    if (!t->is_symbolic_const() || t->get_sort()->get_sort_kind() != BOOL)
```

Comment:
Just FYI, we relaxed this requirement recently. See the reasoning here: https://github.com/makaimann/smt-switch/pull/157. Basically the TLDR is that through the SMT-LIB interface you can use a `define-fun` without creating a fresh literal as your assumption, and so through the API it makes sense to allow any boolean formula. Only mathsat has the restriction that it needs to be a boolean literal (as far as I know).